### PR TITLE
Add Logic App Container Pipelines

### PR DESCRIPTION
**Pull request by Bec-Lyons**
Adds pipelines to build and deploy Logic Apps as containers. 

@liliankasem @derek1ee

### CODE REVIEW

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+env:
+  AZURE_LOGICAPP_PACKAGE_PATH: '.'
+  DOTNET_VERSION: '3.1.301'    # set this to the dotnet version to use
+  RG_LA: 'your-la-rg'
+  RG_CON: 'your-con-rg'
+  LA_NAME: 'your-la-name'
+  Cont_Image: 'your-cont-image'
+  Cont_Tag: 'your-tag'
+  acrUsername: 'your-acr-name'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:14]: Naming here isn't really consistent

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      uses: azure/docker-login@v1
+      with:
+        login-server: ${{ env.acrUsername }}.azurecr.io
+        username: ${{ env.acrUsername }}
+        password: ${{secrets.ACRPASSWORD}}
+
+    - name: 'Compose Customized Docker Image'
+      shell: bash
+      run: |
+        docker build . -t  ${{ env.acrUsername }}.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:51]:
```suggestion
        docker build . -t  ${{ env.IMAGE_URI }}
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        docker push  ${{ env.acrUsername }}.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:52]:
```suggestion
        docker push  ${{ env.IMAGE_URI }}
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: 'Run Azure Functions Container Action'
+      uses: Azure/functions-container-action@v1
+      id: fa
+      with:
+        app-name: ${{ env.LA_NAME }}
+        image: '${{ env.acrUsername }}''.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:59]:
```suggestion
        image: '${{ env.IMAGE_URI }}'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        username: ${{ env.acrUsername }}
+        password: ${{secrets.ACRPASSWORD}}
+
+    - name: 'Compose Customized Docker Image'
+      shell: bash
+      run: |
+        docker build . -t  ${{ env.acrUsername }}.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}
+        docker push  ${{ env.acrUsername }}.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}
+
+    - name: 'Run Azure Functions Container Action'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:54]:
```suggestion
    - name: 'Deploy Logic App Container'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        azPSVersion: latest
+
+    - name: 'Docker Login'
+      uses: azure/docker-login@v1
+      with:
+        login-server: ${{ env.acrUsername }}.azurecr.io
+        username: ${{ env.acrUsername }}
+        password: ${{secrets.ACRPASSWORD}}
+
+    - name: 'Compose Customized Docker Image'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:48]:
```suggestion
    - name: 'Build & Push Docker Image'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+     #build container image
+    - name: Generate Connections
+      uses: azure/powershell@v1
+      with:
+        inlineScript: |
+          ./Scripts/Generate-Connections.ps1 -resourceGroup '${{ env.RG_CON }}'
+        azPSVersion: latest
+
+    - name: 'Docker Login'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:41]:
```suggestion
    - name: 'Login to Azure Container Registry'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 8]
```diff
@@ -0,0 +1,62 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/azure/functions-action
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+
+name: (container) Build and Deploy Logic App. 
+
+env:
+  AZURE_LOGICAPP_PACKAGE_PATH: '.'
+  DOTNET_VERSION: '3.1.301'    # set this to the dotnet version to use
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:8]: I don't think this is still used?
```suggestion
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 1 above)
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:14]: Let's reduce the duplication
```suggestion
  acrUsername: 'your-acr-name'
  IMAGE_URI: '${{ env.acrUsername }}.azurecr.io/${{ env.Cont_Image }}:${{ env.Cont_Tag }}'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 9]
```diff
@@ -0,0 +1,62 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/azure/functions-action
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+
+name: (container) Build and Deploy Logic App. 
+
+env:
+  AZURE_LOGICAPP_PACKAGE_PATH: '.'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:7]: I don't think this is still used?

```suggestion
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  DOTNET_VERSION: '3.1.301'    # set this to the dotnet version to use
+  RG_LA: 'your-la-rg'
```
Comment (tomkerkhove) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:9]: I don't think this is still used?


```suggestion
  RG_LA: 'your-la-rg'
```

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 1 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 3 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 5 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 7 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 4 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 8 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml (hunk 1 above)
Comment (Bec-Lyons) [reply github-sample/.github/workflows/logicapp_container_deploy.yml]: Resolved in latest commit. Thanks.

File: github-sample/.github/workflows/logicapp_container_deploy.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Docs for the Azure Web Apps Deploy action: https://github.com/azure/functions-action
+# More GitHub Actions for Azure: https://github.com/Azure/actions
+
+name: (container) Build and Deploy Logic App. 
+
+env:
+  RG_CON: 'your-con-rg'
+  LA_NAME: 'your-la-name'
+  IMAGE_NAME: 'your-image-name'
+  IMAGE_TAG: 'your-tag'
```
Comment (liliankasem) [inline github-sample/.github/workflows/logicapp_container_deploy.yml:10]: I would suggest using the build id of the pipeline run as the image tag, but not a blocker

File: github-sample/ARM/connectors-parameters.json [hunk 12]
```diff
@@ -9,7 +9,7 @@
             "value": "__2__"
         },
         "sa_name": {
-            "value": "consa"
+            "value": "your-sa-name"
```
Comment (liliankasem) [inline github-sample/ARM/connectors-parameters.json:12]: nit: rename sa variable, not clear what it stands for

File: github-sample/.github/workflows/IaC_container_deploy.yml [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+      uses: Azure/arm-deploy@v1.0.1
+      with:
+        scope: resourcegroup  
+        subscriptionId: ${{ secrets.AZURE_SUB }}
+        resourceGroupName: ${{ env.RG_CON }}
+        template: ARM/connectors-template.json  # Set this to the location of your template file
+        parameters: ARM/connectors-parameters.json logicAppSystemAssignedIdentityTenantId=${{ steps.acrdeploy.outputs.logicAppSystemAssignedIdentityTenantId }} logicAppSystemAssignedIdentityObjectId=${{ steps.acrdeploy.outputs.logicAppSystemAssignedIdentityObjectId }}
+        deploymentMode: Incremental
+
+    #build container image
```
Comment (liliankasem) [inline github-sample/.github/workflows/IaC_container_deploy.yml:58]: I think to keep this consistent to the non-container IaC deployment, should we get rid of the build, push  & deploy steps

**Review by liliankasem — APPROVED** [review]

**Pull request merged**